### DISCUSSION TITLE
Licensing TokoBapak

### DISCUSSION THREAD

**Discussion by aldy505** [TokoBapak/.github#16] [General]
TokoBapak adalah satu project yang jelas-jelas public dan bisa dilihat source codenya. Tujuan diskusi ini adalah mencari cara terbaik untuk melindungi kode TokoBapak agar tidak di abuse secara komersil. Kasus ini disebut sebagai "free-rider", dimana terdapat suatu pihak yang mencari keuntungan dari hasil pekerjaan orang lain tanpa memberi imbal balik (berupa uang atau apapun) kepada pembuat tersebut.

Terdapat suatu usaha dari tim infrastructure untuk menyusahkan siapapun yang ingin melakukan deployment TokoBapak untuk menanggulangi masalah ini. Namun tentu saja itu tidak cukup, ada beberapa effort terkait legal yang harus dilakukan, dalam wujud lisensi.

Terdapat beberapa opsi terkait lisensi yang cocok untuk use case TokoBapak:
- Business Source License (BUSL 1.1) - https://mariadb.com/bsl11/
- Functional Source License (FSL) - https://fsl.software/

Kedua opsi tersebut serupa, dimana:
- Pengguna dilarang untuk menjual atau mengkomersilkan source code TokoBapak
- Pengguna dilarang untuk menjadi kompetitor (menjalankan bisnis) dengan menggunakan source code TokoBapak
- Setelah 4 tahun (atau 2 tahun untuk FSL), lisensi akan berubah menjadi Apache-2.0 (atau MIT) dan baru boleh dijual secara komersil.

Ada research tentang kasus ini, sebagai [Delayed Open Source Publication](https://opensource.org/delayed-open-source-publication/).

Diskusi ini berharap kita sebagai tim TokoBapak bisa setuju tentang lisensi apa yang akan digunakan serta berbagai pertimbangan dan considerationnya.

**Comment by WahidinAji** [comment]
saya baca dulu sambil masak mie ya pak
